consumers/google.go

```diff
@@ -128,6 +129,33 @@ func (d *googleDNSConsumer) Sync(endpoints []*pkg.Endpoint) error {
 	return nil
 }
 
+func (d *googleDNSConsumer) Consume(endpoints chan *pkg.Endpoint, errors chan error, done chan struct{}, wg *sync.WaitGroup) {
```

Comment:
here and likewise Consume methods, might make sense to pass in directed channel, i.e. 

```
func (d *googleDNSConsumer) Consume(endpoints <-chan *pkg.Endpoint,...
```

Likewise in monitoring, only channel u can push to:

```
func (a *fakeProducer) Monitor(results chan<- *pkg.Endpoint
```